library/azure_rm_virtualmachinescalesetinstance.py

```diff
@@ -0,0 +1,184 @@
+            - The name of the VM scale set.
+        required: True
+    instance_id:
+        description:
+            - The instance ID of the virtual machine.
+        required: True
+    latest_model:
+        type: bool
+        description:
+            - Set to C(yes) to upgrade to the latest model.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Upgrade instance to the latest image
+    azure_rm_computevirtualmachinescalesetinstance:
+      resource_group: myResourceGroup
+      vmss_name: myVMSS
+      instance_id: "2"
+      latest_model: yes
+'''
+
+RETURN = '''
+instances:
+    description: A list of instances.
+    returned: always
+    type: complex
+    contains:
+        instance_id:
+            description:
+                - Virtual Machine instance Id
+            returned: always
+            type: str
+            sample: 0
+        tags:
+            description:
+                - Resource tags
+            returned: always
+            type: complex
+            sample: { 'tag1': 'abc' }
+        latest_model:
+            description:
+                - Is latest model applied?
+            returned: always
+            type: bool
+            sample: True
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.compute import ComputeManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMVirtualMachineScaleSetInstance(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            vmss_name=dict(
+                type='str',
+                required=True
+            ),
+            instance_id=dict(
+                type='str'
+            ),
+            latest_model=dict(
+                type='bool'
+            )
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False
+        )
+        self.mgmt_client = None
+        self.resource_group = None
+        self.vmss_name = None
+        self.instance_id = None
+        self.latest_model = None
+        super(AzureRMVirtualMachineScaleSetInstance, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+        self.mgmt_client = self.get_mgmt_svc_client(ComputeManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        self.results['instances'] = self.get()
+
+        self.results['aaa'] = (self.latest_model is not None)
```

Comment:
better naming?

Reply:
sorry, that was mistake, just debugging code ;-)